service/datasetworker/scan.go

```diff
@@ -2,191 +2,14 @@ package datasetworker
 
 import (
 	"context"
-	"strconv"
 
-	"github.com/data-preservation-programs/singularity/database"
 	"github.com/data-preservation-programs/singularity/handler/datasource"
 	"github.com/data-preservation-programs/singularity/model"
-	"github.com/pkg/errors"
-	"github.com/rjNemo/underscore"
 )
 
 // scan scans the data source and inserts the packJobing strategy back to database
 // scanSource is true if the source will be actually scanned in addition to just picking up remaining ones
 // resume is true if the scan will be resumed from the last scanned file, which is useful for resuming a failed scan
 func (w *Thread) scan(ctx context.Context, source model.Source, scanSource bool) error {
-	db := w.dbNoContext.WithContext(ctx)
-	directoryCache := make(map[string]uint64)
-	dataset := *source.Dataset
-	var remaining = newRemain()
-	var remainingFileRanges []model.FileRange
-	err := db.Joins("File").
-		Where("source_id = ? AND file_ranges.pack_job_id is null", source.ID).
-		Order("file_ranges.id asc").
-		Find(&remainingFileRanges).Error
-	if err != nil {
-		return err
-	}
-	w.logger.With("remaining", len(remainingFileRanges)).Info("remaining file ranges")
-	remaining.add(remainingFileRanges)
-
-	if !scanSource {
-		for len(remaining.fileRanges) > 0 {
-			err = w.packJobOnce(ctx, source, dataset, remaining)
-			if err != nil {
-				return errors.Wrap(err, "failed to save packJobing")
-			}
-		}
-		return nil
-	}
-
-	sourceScanner, err := w.datasourceHandlerResolver.Resolve(ctx, source)
-	if err != nil {
-		return errors.Wrap(err, "failed to get source scanner")
-	}
-	entryChan := sourceScanner.Scan(ctx, "", source.LastScannedPath)
-	for entry := range entryChan {
-		if entry.Error != nil {
-			w.logger.Errorw("failed to scan", "error", entry.Error)
-			continue
-		}
-
-		file, fileRanges, err := datasource.PushFile(ctx, w.dbNoContext, entry.Info, source, dataset, directoryCache)
-		if err != nil {
-			return errors.Wrap(err, "failed to push file")
-		}
-		if file == nil {
-			w.logger.Infow("file already exists", "path", entry.Info.Remote())
-			continue
-		}
-		err = database.DoRetry(ctx, func() error {
-			return db.Model(&model.Source{}).Where("id = ?", source.ID).
-				Update("last_scanned_path", file.Path).Error
-		})
-		if err != nil {
-			return errors.Wrap(err, "failed to update last scanned path")
-		}
-
-		remaining.add(fileRanges)
-		for remaining.carSize >= dataset.MaxSize {
```

Comment:
I believe you've removed this line after refactoring.
This logic is quite important because it can take days to scan a datasource and we want to create packJob while the data source is being scanned.
So whenever the remaining size is larger than a sector size

Reply:
I don't this is accurate. We just do it differently. I've referenced lines where we handle this.

Reply:
I think we're talking about two different things.

I'll comment on the lines on the new file